WooCommerce/Classes/ViewRelated/CardPresentPayments/CardPresentPaymentsModalViewController.swift

```diff
@@ -277,14 +279,29 @@ private extension CardPresentPaymentsModalViewController {
     }
 
     func configureAuxiliaryButton() {
+        clearAuxiliaryButton()
+
         guard shouldShowAuxiliaryButton() else {
             auxiliaryButton.isHidden = true
             return
         }
 
-        auxiliaryButton.setTitleWithoutAnimation(viewModel.auxiliaryButtonTitle, for: .normal)
         auxiliaryButton.isHidden = false
+        auxiliaryButton.setTitle(viewModel.auxiliaryButtonTitle, for: .normal)
         auxiliaryButton.accessibilityIdentifier = Accessibility.auxiliaryButton
+
+        if viewModel.actionsMode == .secondaryActionAndAttributedAuxiliaryButton {
+            auxiliaryButton.setImage(.infoOutlineImage, for: .normal)
+            auxiliaryButton.setAttributedTitle(viewModel.auxiliaryAttributedButtonTitle, for: .normal)
+            auxiliaryButton.distributeTitleAndImage(spacing: 12.0)
+        }
```

Comment:
I don't think there's anything about the `actionsMode = .secondaryActionAndAttributedAuxiliaryButton` that logically says you should set an image and an attributed title here.

I'd suggest that you set them to the value in the viewModel (and put the image on the viewModel) regardless of whether they're nil or populated. You could put the `distributeTitleAndImage` call behind a check for `if viewModel.auxiliaryButtonImage == nil` if that would help.

Reply:
My main concern when I initially tried to make these changes was that the viewModel used here is not specifically our `CardPresentModalScanningForReader`, but the `CardPresentPaymentsModalViewModel` protocol, so I would need a way to filter what content I'm loading. 

I could use something like `viewModel as? CardPresentModalScanningForReader != nil` for the check, but this feels wrong as would compile but may not result on what we want on runtime, also would not resolve the issue of `auxiliaryButtonImage` not being part of the general viewModel, so I would need to add another default implementation for `auxiliaryButtonImage` to the private extension, as I already had to do with `auxiliaryAttributedButtonTitle`.

Happy to go through this part on another session, I may be miss-understanding how to filter down this properly 👍